exporter/elasticsearchexporter/esclient.go

```diff
@@ -145,7 +150,7 @@ func newElasticsearchClient(
 		RetryOnError: func(_ *http.Request, err error) bool {
 			return !errors.Is(err, context.Canceled) && !errors.Is(err, context.DeadlineExceeded)
 		},
-		MaxRetries:   min(defaultMaxRetries, config.Retry.MaxRetries),
```

Comment:
Yikes, I don't know what I was thinking. Can we add a unit test that would have failed, so I don't make the same stupid mistake twice? :sweat_smile: